File: 구름 알고리즘 미션/기초 알고리즘 챌린지/구름 RPG 1/구름 RPG 1.js
// Run by Node.js
const readline = require('readline');

(async () => {
	let rl = readline.createInterface({ input: process.stdin });
	let input = [];
	for await (const line of rl) {
		input.push(line);
		if(input.length === +input[0] + 1) rl.close();
	}
	
	const isPrime = (n) => {
		for(let i = 2 ; i < parseInt(Math.sqrt(n)) + 1 ; i++){
			if(n % i === 0) return false;
		}
		return true;
	}
	const N = +input[0];
	for(let i = 1 ; i <= N ; i++){
		const A = +input[i];
		console.log(isPrime(A) ? "Yes" : "No");
	}
	
	process.exit();
})();

File: 구름 알고리즘 미션/기초 알고리즘 챌린지/구름 RPG 1/구름 RPG 1.py
def isPrime(n):
	for i in range(2, int(n ** (1/2)) + 1):
		if n % i == 0:
			return False
	return True

N = int(input())

for _ in range(N):
	A = int(input())
	print("Yes" if isPrime(A) else "No")